espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -160,6 +162,44 @@ def reset_parameters(self, args):
         # initialize parameters
         initialize(self, args.transformer_init)
 
+    def wfst_based_decoding(self, x, HLG, recog_args, char_list=None):
+        """wfst-based decoding.
+
+        :param ndnarray x: input acoustic feature (B, T, D) or (T, D)
+        :param Namespace recog_args: argment Namespace contraining options
+        :param list char_list: list of characters
+        :return: decoding result
+        :rtype: list
+        """
+        self.eval()
+        encoder_out, encoder_mask = self.encoder(x, None)
+        log_probs = self.ctc.log_softmax(
+            encoder_out)
+        log_probs[:, :, 0] += -1
+        dense_fsa_vec_seg = torch.stack(
+             (torch.tensor(range(encoder_out_lens.shape[0])),
```

Comment:
Where is `encoder_out_lens` defined? I don't see the line where `encoder_out_lens` is defined.